browser/src/map/handler/Map.Keyboard.js

```diff
@@ -462,6 +464,21 @@ L.Map.Keyboard = L.Handler.extend({
 			return;
 		}
 
+		// save-as for German shortcuts - F12.
+		if (ev.type === 'keydown' && !ev.altKey && !this.modifier && keyCode === this.keyCodes.F12 && app.UI.language.fromURL === 'de') {
+			if (this._map.uiManager.getCurrentMode() === 'notebookbar') {
+				this._map.openSaveAs(); // Opens save as dialog if integrator supports it.
+				ev.preventDefault();
+				return;
+			}
+		}
+
+		if (ev.type === 'keydown' && !ev.altKey && ev.shiftKey && keyCode === this.keyCodes.F3 && app.UI.language.fromURL === 'de') {
+			app.socket.sendMessage('uno .uno:ChangeCaseRotateCase');
```

Comment:
best to use high level map.sendUnoCommand  https://github.com/CollaboraOnline/online/blob/5520965b151cb854acbac3f1bbb68aec5ac2e1f2/browser/src/control/Toolbar.js#L343

because it handles other things: like check if allowed in readonly mode, blink dialog if other is already opened etc...